components/modal/ModalMediaPreviewCarousel.vue

```diff
@@ -17,88 +15,222 @@ const { modelValue } = defineModels<{
+    modelValue.value = Math.min(media.length - 1, modelValue.value + 1)
+  if (vert === -1) // top
+    emit('close')
+
+  goToFocusedSlide()
+}
+
+function slideToClosestSlide() {
+  const startOfFocusedSlide = slide.value[modelValue.value].offsetLeft * scale.value
+  const slideWidth = slide.value[modelValue.value].offsetWidth * scale.value
+
+  if (x.value > startOfFocusedSlide + slideWidth / 2)
+    modelValue.value = Math.min(media.length - 1, modelValue.value + 1)
+  else if (x.value < startOfFocusedSlide - slideWidth / 2)
+    modelValue.value = Math.max(0, modelValue.value - 1)
+
+  goToFocusedSlide()
+}
+
+function handleDrag(delta: [number, number], movement: [number, number]) {
+  isDragging.value = true
+
+  if (isZoomedIn.value)
+    handleZoomDrag(delta)
+  else
+    handleSlideDrag(movement)
+}
+
+function handleZoomDrag([deltaX, deltaY]: [number, number]) {
+  x.value -= deltaX / scale.value
+  y.value -= deltaY / scale.value
+
+  restrictDragToInsideSlide()
+}
+
+function handleSlideDrag([movementX, movementY]: [number, number]) {
+  goToFocusedSlide()
+
+  if (-movementY > Math.abs(movementX)) // upwards movement is more then horizontal
+    y.value -= movementY / scale.value
+  else
+    x.value -= movementX / scale.value
+
+  y.value = Math.max(0, y.value)
+
+  if (media.length === 1)
+    x.value = 0
+}
+
+function restrictDragToInsideSlide() {
+  x.value = Math.min(dragRestrictions.value.right, Math.max(dragRestrictions.value.left, x.value))
+  y.value = Math.min(dragRestrictions.value.bottom, Math.max(dragRestrictions.value.top, y.value))
+}
+
+const sliderStyle = computed(() => {
+  const style = {
+    transform: `scale(${scale.value}) translate(${-x.value}px, ${-y.value}px)`,
+    transition: 'none',
+    gap: `${slideGap}px`,
+  }
+
+  if (canAnimate.value && !isDragging.value && !isPinching.value)
+    style.transition = 'all 0.3s ease'
 
-  return (lengthY.value / height.value) * 100 * -1
+  return style
 })
 </script>
 
 <template>
-  <div ref="target" flex flex-row max-h-full max-w-full overflow-hidden>
-    <div flex :style="{ transform: `translateX(${distanceX}%) translateY(${distanceY}%)`, transition: isSwiping ? 'none' : canAnimate ? 'all 0.5s ease' : 'none' }">
-      <div v-for="item in media" :key="item.id" p4 select-none w-full flex-shrink-0 flex flex-col items-center justify-center>
-        <img max-h-full max-w-full :draggable="false" select-none :src="item.url || item.previewUrl" :alt="item.description || ''">
+  <div ref="view" flex flex-row h-full w-full overflow-hidden>
+    <div ref="slider" :style="sliderStyle" w-full h-full flex items-center>
+      <div
+        v-for="item in media"
+        :key="item.id"
+        ref="slide"
+        flex-shrink-0
+        w-full
+        h-full
+        flex
+        items-center
+        justify-center
+      >
+        <img
+          ref="image"
+          select-none
+          max-w-full
+          max-h-full
+          cursor-pointer
```

Comment:
maybe change `cursor-pointer` to `cursor-grab` and change to `cursor-grabbing` when dragging? on desktop `cursor-pointer` looks weird...

Reply:
maybe cursor shouldn't be changed in the image...

Reply:
Thanks @userquin I think we should signal that the image can be grabbed with the cursor, so I added your first idea :)